### PR TITLE
REALMC-9934: Include error message in Segment event properties

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

File: internal/telemetry/event.go [hunk 1]
```diff
@@ -34,16 +34,23 @@ const (
...
+	eventDataKeyVersion      = "v"
+	eventDataKeyError        = "err"
+	eventDataKeyErrorMessage = "err_msg"
 
 	// EventDataKeyTemplate used to tracked if templates were used to create an app
 	EventDataKeyTemplate = "templateId"
 )
 
+func EventDataError(err error) []EventData {
+	return []EventData{
```
Comment (tsedgwick) [inline internal/telemetry/event.go:48]: should we check `if err != nil`? This will panic otherwise

**Review by tsedgwick — APPROVED** [review]
one question - otherwise LGTM

also, the PR title should be REALMC-9934 (missing the C)

File: internal/telemetry/event.go (hunk 1 above)
Comment (makesitgo) [reply internal/telemetry/event.go:48]: the usages of this function are already wrapped in `!= nil`, but the better-safe-than-sorry approach seems fine here

**Comment by makesitgo** [comment]
was able to confirm with sumedha this is working as expected now, it's good to go in whatever next release we have

**Pull request merged**